src/uct/tcp/tcp_listener.h

```diff
@@ -0,0 +1,45 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2019.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include "tcp_sockcm.h"
+#include <uct/base/uct_cm.h>
+
+/**
+ * An TCP listener for incoming connections requests on the server side.
+ */
+typedef struct uct_tcp_listener {
+    uct_listener_t                       super;
+
+    int                                  listener_fd;
+
+    /** Callback to invoke upon receving a connection request from a client */
+    uct_listener_conn_request_callback_t conn_request_cb;
+
+    /** User's data to be passed as argument to the conn_request_cb */
+    void                                 *user_data;
+
+    /** List of open file descriptors. One per client */
+    ucs_list_link_t                      fd_list;
+} uct_tcp_listener_t;
+
+
+typedef struct uct_tcp_listener_ctx {
+    int              fd;
+    ucs_list_link_t  list;
+} uct_tcp_listener_ctx_t;
```

Comment:
where is no any link to listener object, maybe rename to `uct_tcp_listen_ctx_t` or `uct_tcp_accept_ctx_t`?

Reply:
i didn't want to add something that isn't used in this pr.
since it's a struct in the tcp_listener file, i think it's better to call it uct_tcp_listener_ctx_t

Reply:
structure/type name should describe its meaning, not file where it's defined

Reply:
i agree. here it does both - this is the context that the tcp listener is using.

Reply:
I don't see any link to listener here...

Reply:
it is used on the tcp listener.